Shared/Tests.Shared/NotificationTests.cs

```diff
@@ -252,6 +275,143 @@ public void CollectionChanged_WhenTransactionHasBothAddAndRemove_ShouldReset()
             }
         }
 
+        [Test]
+        public void List_WhenUnsubscribed_ShouldStopReceivingNotifications()
+        {
+            var container = new OrderedContainer();
+            _realm.Write(() => _realm.Add(container));
+
+            var eventArgs = new List<NotifyCollectionChangedEventArgs>();
+            var handler = new NotifyCollectionChangedEventHandler((sender, e) =>
+            {
+                eventArgs.Add(e);
+            });
+
+            var collection = container.Items.AsRealmCollection();
+            collection.CollectionChanged += handler;
+
+            _realm.Write(() =>
+            {
+                container.Items.Add(new OrderedObject());
+            });
+
+            TestHelpers.RunEventLoop();
+
+            Assert.That(eventArgs.Count, Is.EqualTo(1));
+            Assert.That(eventArgs[0].Action, Is.EqualTo(NotifyCollectionChangedAction.Add));
+
+            collection.CollectionChanged -= handler;
+
+            _realm.Write(() =>
+            {
+                container.Items.Add(new OrderedObject());
+            });
+
+            TestHelpers.RunEventLoop();
+
+            Assert.That(eventArgs.Count, Is.EqualTo(1));
+            Assert.That(eventArgs[0].Action, Is.EqualTo(NotifyCollectionChangedAction.Add));
+        }
+
+        [Test]
+        public void List_WhenTransactionHasBothAddAndRemove_ShouldReset()
+        {
+            var container = new OrderedContainer();
+            container.Items.Add(new OrderedObject());
+            _realm.Write(() => _realm.Add(container));
+
+            var eventArgs = new List<NotifyCollectionChangedEventArgs>();
+            var collection = container.Items.AsRealmCollection();
+            collection.CollectionChanged += (sender, e) =>
+            {
+                eventArgs.Add(e);
+            };
+
+            _realm.Write(() =>
+            {
+                container.Items.Clear();
+                container.Items.Add(new OrderedObject());
+            });
+
+            TestHelpers.RunEventLoop();
+
+            Assert.That(eventArgs.Count, Is.EqualTo(1));
+            Assert.That(eventArgs[0].Action, Is.EqualTo(NotifyCollectionChangedAction.Reset));
+        }
+
+        [TestCaseSource(nameof(CollectionChangedTestCases))]
+        public void TestRealmListNotifications(int[] initial, NotifyCollectionChangedAction action, int[] change, int startIndex)
+        {
+            var container = new OrderedContainer();
+            foreach (var i in initial)
+            {
+                container.Items.Add(new OrderedObject
```

Comment:
It's not a rule, I could have written it as `container.Items.Add(new OrderedObject { Order = i });`